mdbenchmark/plot.py

```diff
@@ -0,0 +1,210 @@
+    ax.legend()
+
+    ax2 = ax.twiny()
+    ax1_xticks = ax.get_xticks()
+    ax2.set_xticks(ax1_xticks)
+    ax2.set_xticklabels(gb.get_group(list(gb.groups.keys())[0])[df_sel])
+    ax2.set_xbound(ax.get_xbound())
+    #ax2.set_xlabel('{}'.format('{}\n\nCores'.format(df['host'][0])))
+
+    return ax
+
+
+@cli.command()
+@click.option(
+    '--csv',
+    help='name of csv file',
+    multiple=True,
+    show_default=True)
+@click.option(
+    '--output-name',
+    '-o',
+    default=None,
+    help="name of output files",
+    show_default=True)
+@click.option(
+    '--output-type',
+    '-t',
+    help="file extension for plot outputs",
+    type=click.Choice(['png', 'pdf', 'svg', 'jpeg']),
+    show_default=True,
+    default='png')
+@click.option(
+    '--host-name',
+    '-h',
+    default=None,
+    multiple=True,
+    help="hostname",
+    show_default=True)
+@click.option(
+    '--module-name',
+    '-h',
+    default=None,
+    multiple=True,
+    help="module name or engine name (gromacs, namd )",
+    show_default=True)
+@click.option(
+    '--host-name',
+    '-h',
+    default=None,
+    multiple=True,
+    help="hostname",
+    show_default=True)
+@click.option(
+    '--gpu/--no-gpu',
+    help="plot data for GPU runs",
+    show_default=True,
+    default=False)
+@click.option(
+    '--cpu/--no-cpu',
+    help="plot data for GPU runs",
+    show_default=True,
+    default=True)
+@click.option(
+    '--plot-nodes/--plot-cores',
+    help="Plot performance per node instead of core",
+    show_default=True,
+    default=False)
+def plot(csv, output_name, output_type, host_name, module_name, gpu, cpu, plot_nodes):
+    """Plot nice things"""
+    print(plot_nodes)
+    df_list = []
+    for c in csv:
+        tmp_df = pd.read_csv(c, index_col=0)
+        # append df_list
+        df_list.append(tmp_df)
+
+    df = pd.concat(df_list)
+    # Remove NaN values. These are missing ncores/performance data.
+    df = df.dropna()
+
+    # preprocess the commandline entries
+
+    df_module_list = df['module'].tolist()
+    processed_module_names = []
+    print(df_module_list)
+    for module in module_name:
+        if module in ['namd', 'gromacs']:
+            real_module_names = [s for s in df_module_list if module in s]
+        elif module in df_module_list:
+            processed_module_names.append(module)
+        elif module not in df_module_list:
+            console.warn("The module {} does not exist in your data. Exiting",
+                         module)
+    if len(module_name) is not 0:
+        processed_module_names = processed_module_names + real_module_names
+
+    host_list = df['host'].tolist()
+    for host in host_name:
+        if host not in host_list:
+            console.warn("The host {} does not exist in your csv data. Exiting.",
+                         host)
+
+    gpu_list = []
+    if gpu is True:
+        gpu_list.append(True)
+    if cpu is True:
+        gpu_list.append(False)
```

Comment:
Should this be `cpu_list`?

Reply:
we can rename this to cpu_gpu or something. since for gpu or cpu either false or true is listed in the dataframe i make a list containing either one of them or both if the user wants to plot both gpu and cpu